docs/developer-guide/ksqldb-clients/contributing.md

```diff
@@ -0,0 +1,175 @@
+---
+layout: page
+title: Contributing a new ksqlDB client
+tagline: Contributing a new ksqlDB client
+description: Contribute a new ksqlDB client for your favorite language.
+keywords: ksqlDB, client
+---
+
+If you're interested in using ksqlDB from a programming language we don't yet support, we've created
+this guide to make it easy to add your own [ksqlDB client](index.md). Thanks for your contribution!
```

Comment:
Could do with a preamble about the general design of the protocol e.g. 

> Clients communicate with the ksqlDB server via HTTP. Admin requests are submitted using POST requests. Query results and inserts are streaming over HTTP2.  The sections below cover the different types of functionality a client must/could (TBD) support and links to the appropriate API documentation., where example request and responses can be found.

Not really sure if that's correct or not, but you hopefully get the idea: we need to explain to the would be client engineer the general architecture and what to expect in this pge.